core/state/state_object.go

```diff
@@ -355,10 +353,18 @@ func (s *stateObject) updateTrie(db Database) Trie {
 		var v []byte
 		if (value == common.Hash{}) {
 			s.setError(tr.TryDelete(key[:]))
+			s.db.db.TrieDB().DiskDB().Delete(append(s.address[:], key[:]...))
 		} else {
 			// Encoding []byte cannot fail, ok to ignore the error.
 			v, _ = rlp.EncodeToBytes(common.TrimLeftZeroes(value[:]))
-			s.setError(tr.TryUpdate(key[:], v))
+
+			// Update the trie, with v as a value
+			k := trie.GetTreeKeyStorageSlot(s.address, big.NewInt(0).SetBytes(key[:]))
+			s.setError(tr.TryUpdate(k, v))
+
+			// Also save the account data in a location that is easy to
+			// find, i.e. right after the account's key.
+			s.db.db.TrieDB().DiskDB().Put(append(s.address[:], key[:]...), v)
```

Comment:
@s1na this is where the data is written directly to disk after the tree is updated. This is the part that needs to be replaced with the snapshot.